snapshot/android/snapshot-generator.js

```diff
@@ -217,3 +244,134 @@ SnapshotGenerator.prototype.generate = function (options) {
+        const buildCSourceCommand = this.getXxdCommand(srcOutputDirInDocker);
+        // add vim in order to get xxd
+        command = `docker run -v "${blobInputDir}:${blobsInputInDocker}" -v "${srcOutputDir}:${srcOutputDirInDocker}" ${SNAPSHOTS_DOCKER_IMAGE} /bin/sh -c "cd ${blobsInputInDocker} && apk add vim && ${buildCSourceCommand}"`;
+    }
+    else {
+        command = this.getXxdCommand(srcOutputDir);
+    }
+    shellJsExecuteInDir(blobInputDir, function () {
+        shelljs.exec(command);
+    });
+}
+
+SnapshotGenerator.prototype.runMksnapshotTool = function (tool, mksnapshotParams, inputFile, snapshotInDocker, allToolsDir, buildCSource) {
+    const toolPath = tool.path;
+    const androidArch = this.convertToAndroidArchName(tool.arch);
+    if (!fs.existsSync(toolPath)) {
+        throw new Error("Can't find mksnapshot tool for " + androidArch + " at path " + toolPath);
+    }
+
+    const tempFolders = [];
+    return new Promise((resolve, reject) => {
+        console.log("***** Generating snapshot for " + androidArch + " *****");
+        const inputFileDir = dirname(inputFile);
+        const blobOutputDir = join(this.buildPath, "snapshots/blobs", androidArch);
+        createDirectory(blobOutputDir);
+        const toolParams = mksnapshotParams || "--profile_deserialization";
+
+        let command = "";
+        if (snapshotInDocker) {
+            this.setupDocker();
+            const appDirInDocker = "/app";
+            const blobOutputDirInDocker = `/dist/blobs/${androidArch}`;
+            const toolsTempFolder = join(inputFileDir, "tmp");
+            tempFolders.push(toolsTempFolder);
+            const toolPathInAppDir = this.copySnapshotTool(allToolsDir, toolPath, toolsTempFolder);
+            const toolPathInDocker = this.getPathInDocker(inputFileDir, appDirInDocker, toolPathInAppDir);
+            const inputFilePathInDocker = this.getPathInDocker(inputFileDir, appDirInDocker, inputFile);
+            const outputPathInDocker = this.getPathInDocker(blobOutputDir, blobOutputDirInDocker, blobOutputDir);
+            const toolCommandInDocker = this.getSnapshotToolCommand(toolPathInDocker, inputFilePathInDocker, outputPathInDocker, toolParams);
+            command = `docker run -v "${inputFileDir}:${appDirInDocker}" -v "${blobOutputDir}:${blobOutputDirInDocker}" ${SNAPSHOTS_DOCKER_IMAGE} /bin/sh -c "${toolCommandInDocker}"`;
+        } else {
+            command = this.getSnapshotToolCommand(toolPath, inputFilePath, outputPath, toolParams);
+        }
+
+        // Generate .blob file
+        child_process.exec(command, { encoding: "utf8" }, (error, stdout, stderr) => {
+            tempFolders.forEach(tempFolder => {
+                shelljs.rm("-rf", tempFolder);
+            });
+
+            const snapshotError = this.handleSnapshotToolResult(error, stdout, stderr, androidArch);
+            if (snapshotError) {
+                return reject(error);
```

Comment:
Shouldn't we reject here with `snapshotError` as we've built it with `errorHeader` and `errorFooter`?
```suggestion
                return reject(snapshotError);
```